flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/write/committer/HadoopRenameFileCommitter.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connectors.hive.write.committer;
+
+import org.apache.flink.connectors.hive.write.HadoopFileCommitter;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import java.io.IOException;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/**
+ * The Hadoop file committer that directly rename the in-progress file
+ * to the target file. For FileSystem like S3, renaming may lead to
+ * additional copies.
+ */
+public class HadoopRenameFileCommitter implements HadoopFileCommitter {
+
+	private final Configuration configuration;
+
+	private final Path targetPath;
+
+	private final Path inProgressFilePath;
+
+	public HadoopRenameFileCommitter(Configuration configuration, Path targetPath) {
+		this.configuration = configuration;
+		this.targetPath = targetPath;
+		this.inProgressFilePath = generateInProgressFilePath();
+	}
+
+	@Override
+	public Path getTargetPath() {
+		return targetPath;
+	}
+
+	@Override
+	public Path getInProgressFilePath() {
+		return inProgressFilePath;
+	}
+
+	@Override
+	public void preCommit() {
+		// Do nothing.
+	}
+
+	@Override
+	public void commit() throws IOException {
+		FileSystem fileSystem = FileSystem.get(targetPath.toUri(), configuration);
+
+		try {
+			fileSystem.rename(inProgressFilePath, targetPath);
+		} catch (IOException e) {
+			throw new IOException(
+				String.format("Could not commit file from %s to %s", inProgressFilePath, targetPath),
+				e);
+		}
+	}
+
+	@Override
+	public void commitAfterRecovery() throws IOException {
```

Comment:
`commitAfterRecovery` is same to `commit`?

Reply:
`commitAfterRecovery` is not the same with `commit` since  `commitAfterRecovery` is expected to be called after failover, at this time the file might be committed or partially committed, thus `commitAfterRecovery` should do some more checking with the status.

Reply:
```
private void rename(boolean checkFileExists) throws IOException {
		FileSystem fileSystem = FileSystem.get(targetPath.toUri(), configuration);

		if (fileSystem.exists(inProgressFilePath)) {
			try {
				// If file exists, it will be overwritten.
				fileSystem.rename(inProgressFilePath, targetPath);
			} catch (IOException e) {
				throw new IOException(
						String.format("Could not commit file from %s to %s", inProgressFilePath, targetPath),
						e);
			}
		} else {
			throw new IOException(String.format("In progress file(%s) not exists.", inProgressFilePath));
		}
	}
```
Refactor to above?

Reply:
Extracted the `rename` function